File: TestWeatherApp.Api/Core/Repositories/WeatherService.cs
using TestWeatherApp.Api.Core.Models;

namespace TestWeatherApp.Api.Core.Repositories
{
    public class WeatherService : IWeatherService
    {

        private readonly IWeatherRepository _weatherRepository;

        public WeatherService(IWeatherRepository weatherRepository)
        {
            _weatherRepository = weatherRepository;
        }

        public Weather GetWeather(string country, string city)
        {
            return _weatherRepository.GetWeather(country, city);


        }
    }
}

File: TestWeatherApp.Angular/src/app/weather.service.ts
import { Injectable } from '@angular/core';
import { Http, Headers, Response } from '@angular/http';
import { Observable } from 'rxjs/Observable';
import 'rxjs/add/operator/map';

import { environment } from '../environments/environment';

@Injectable()
export class WeatherService {

  apiUrl = environment.apiUrl;
  
  constructor(private _http: Http) {}
   
  getWeather(country: string, city: string): Observable<Response>  {
    return this._http.get(this.apiUrl + '/weather/' + country + '/' + city)
    .map((response: Response) => {
      return response;
    });
   }
      

}
